.github/workflows/universalnrp-test.yml

```diff
@@ -3,127 +3,125 @@ name: Universal NRP Test
-  DEFAULT_PACKAGE_LIST: |
-    [
-      { "name": "LinuxSshServerSecurityBaseline", "short-name": "SSH", "resource-count": 20 },
-      { "name": "AzureLinuxBaseline", "short-name": "ASB", "resource-count": 168 }
-    ]
-
 jobs:
   package:
     name: Package
-    if: ${{ inputs.policyPackageList == '[]' || inputs.policyPackageList == '' }}
+    if: ${{ inputs.policy_packages == '[]' || inputs.policy_packages == '' }}
     uses: ./.github/workflows/package-build.yml
-    strategy:
-      matrix:
-        target:
-          [
-            { os: ubuntu, version: 14.04, arch: amd64, dist: trusty, package-type: DEB },
-          ]
+    needs: [setup-matrix]
     with:
-      target: ${{ matrix.target.os }}-${{ matrix.target.version }}
-      arch: ${{ matrix.target.arch }}
-      artifact: nrp-test-package
-      package-type: ${{ matrix.target.package-type }}
+      target: ubuntu-14.04
+      arch: amd64
+      artifact: nrp-test
+      package-type: DEB
+      test: true
       machine-config: true
       release: ${{ github.event_name == 'pull_request' && false || true }}
 
-  generate-matrix:
-    name: Generate Matrix
-    needs: package
+  setup-matrix:
+    name: Setup Matrix
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.matrix.outputs.matrix }}
-    if: ${{ always() }}
+      targets: ${{ steps.matrix.outputs.targets }}
+      custom_download: ${{ steps.matrix.outputs.custom_download }}
+      policy_packages: ${{ steps.matrix.outputs.policy_packages }}
     steps:
       - name: Generate Matrix
         id: matrix
+        env:
+          DEFAULT_POLICYPACKAGES: |
+            [
+              { "name": "LinuxSshServerSecurityBaseline", "short-name": "SSH", "resource-count": 20 },
+              { "name": "AzureLinuxBaseline", "short-name": "ASB", "resource-count": 168 }
+            ]
+          TARGETS: |
+            [
+              { "os": "centos", "version": "8" },
+              { "os": "debian", "version": "10" },
+              { "os": "debian", "version": "11" },
+              { "os": "debian", "version": "12" },
+              { "os": "mariner", "version": "2" },
+              { "os": "oraclelinux", "version": "8" },
+              { "os": "rhel", "version": "8" },
+              { "os": "rhel", "version": "9" },
+              { "os": "rockylinux", "version": "9" },
+              { "os": "sles", "version": "15" },
+              { "os": "ubuntu", "version": "20.04" },
+              { "os": "ubuntu", "version": "22.04" }
+            ]
+          # { "os": "almalinux", "version": "9" },
+          # { "os": "amazonlinux", "version": "2" },
+          # { "os": "centos", "version": "7" },
+          # { "os": "oraclelinux", "version": "7" },
+          # { "os": "rhel", "version": "7" },
+          # { "os": "sles", "version": "12" },
+          # { "os": "ubuntu", "version": "16.04" },
+          # { "os": "ubuntu", "version": "18.04" },
```

Comment:
nit: inserting them in the TARGETS to keep the  original order would help longer term, rather than separating them here

Reply:
Unfortunately, they need to stay separate since the `TARGETS` env is a JSON string. As I uncomment them and add them back in later PRs I'll make sure to insert them in the correct order.